cvmfs/upload_s3.cc

```diff
@@ -220,13 +217,10 @@ void *S3Uploader::MainCollectResults(void *data) {
   LogCvmfs(kLogUploadS3, kLogDebug, "Upload_S3 WorkerThread started.");
   S3Uploader *uploader = reinterpret_cast<S3Uploader *>(data);
 
-  std::vector<s3fanout::JobInfo *> jobs;
-  while (atomic_read32(&uploader->terminate_) == 0) {
-    jobs.clear();
-    uploader->s3fanout_mgr_->PopCompletedJobs(&jobs);
-    for (unsigned i = 0; i < jobs.size(); ++i) {
+  while (true) {
+    s3fanout::JobInfo *info = uploader->s3fanout_mgr_->PopCompletedJob();
+    if (info) {
```

Comment:
I think it's nicer to reverse the check:
```
if (!info)
  break;
...
```

Reply:
Sure - have updated to reverse the check (and reindented the rest of the loop accordingly).